src/features/musl.ts

```diff
@@ -14,38 +19,55 @@ export async function setUpNativeImageMusl(): Promise<void> {
   }
   let toolPath = tc.find(MUSL_NAME, MUSL_VERSION)
   if (toolPath) {
-    core.info(`Found ${MUSL_NAME} ${MUSL_VERSION} in tool-cache @ ${toolPath}`)
+    core.info(`Found musl ${MUSL_VERSION} in tool-cache @ ${toolPath}`)
   } else {
-    core.startGroup(`Setting up musl for GraalVM Native Image...`)
+    core.startGroup(`Building musl with zlib for GraalVM Native Image...`)
+    // Build musl
+    const muslHome = join(homedir(), 'musl-toolchain')
     const muslDownloadPath = await tc.downloadTool(
-      `https://github.com/graalvm/setup-graalvm/releases/download/x86_64-linux-musl-${MUSL_VERSION}/${MUSL_NAME}.tgz`
+      `https://musl.libc.org/releases/musl-${MUSL_VERSION}.tar.gz`
     )
     const muslExtractPath = await tc.extractTar(muslDownloadPath)
-    const muslPath = join(muslExtractPath, MUSL_NAME)
-
-    const zlibCommit = 'ec3df00224d4b396e2ac6586ab5d25f673caa4c2'
+    const muslPath = join(muslExtractPath, `musl-${MUSL_VERSION}`)
+    const muslBuildOptions = {cwd: muslPath}
+    await exec(
+      './configure',
+      [`--prefix=${muslHome}`, '--static'],
+      muslBuildOptions
+    )
+    await exec('make', [], muslBuildOptions)
+    await exec('make', ['install'], muslBuildOptions)
+    const muslGCC = join(muslHome, 'bin', MUSL_NAME)
+    await fs.symlink(
+      muslGCC,
+      join(muslHome, 'bin', 'x86_64-linux-musl-gcc'),
```

Comment:
So this is still x86_64 only?

Reply:
`musl-gcc` needs to be called `x86_64-linux-musl-gcc` because that's what Native Image expects. Not sure what happens under aarch64, but it's not supported.